engine/runtime-compiler/src/main/java/org/enso/compiler/data/CompilerConfig.java

```diff
@@ -0,0 +1,114 @@
+package org.enso.compiler.data;
+
+import java.io.PrintStream;
+import scala.Option;
+
+/**
+ * Configuration for the compiler.
+ *
+ * @param autoParallelismEnabled whether or not automatic parallelism detection is enabled.
+ * @param warningsEnabled whether or not warnings are enabled
+ * @param privateCheckEnabled whether or not private keyword is enabled
+ * @param staticTypeInferenceEnabled whether or not type inference is enabled
+ * @param treatWarningsAsErrors If warnings should be treated as errors.
+ * @param dumpModuleIR identification (name) of a module to dump
+ * @param isStrictErrors if true, presence of any Error in IR will result in an exception
+ * @param isLintingDisabled if true, compilation should not run any linting passes
+ * @param outputRedirect redirection of the output of warnings and errors of compiler
+ */
+public record CompilerConfig(
```

Comment:
Too bad it is not possible to hide this constructor of `record` and force people to use the `Builder`...